taiga2/controllers/endpoint.py

```diff
@@ -515,6 +525,14 @@ def create_upload_session_file(uploadMetadata, sid):
             data_file = models_controller.get_datafile_by_taiga_id(
                 existing_taiga_id, one_or_none=True
             )
+
+            # When adding a reference to an existing Taiga datafile, it is possible
+            # to provide a new set of custom metadata in the form of key/value pairs.
+            # If new custom_metadata is not provided, copy the key/value pairs from the
+            # original file.
+            if data_file != None and custom_metadata == None:
```

Comment:
You could move the 
```
if custom_metadata == None:
   custom_metadata = data_file.custom_metadata
```

after line 545 and remove the `data_file != None`

It's functionally equivillent to what you have written, but I think it's easier to read when there are fewer conditions checked. (In my mind, the `data_file == None` is an error condition, so I was expecting a check for that, and then all the code that should happen assuming we could find the file.)